src/components/Evolution.ts

```diff
@@ -35,16 +84,23 @@ export class Evolution<TGameStartr extends FullScreenPokemon> extends Component<
      * Holds evolution requirement checks, keyed by the method of evolution.
      */
     private readonly requirementHandlers: IRequirementHandlers = {
```

Comment:
These functions are getting pretty big. I think you should create private member functions of this class and have the handlers call them.